gqcp/tests/QCMethod/CC/QCMethod_CCD_test.cpp

```diff
@@ -0,0 +1,108 @@
+    const auto M = g_spinor_basis.numberOfSpinors();
+    const auto g_sq_hamiltonian = GQCP::SQHamiltonian<double>::Molecular(g_spinor_basis, molecule);  // in the canonical restricted spin-orbitals
+
+    // Create the GHF ONV (which is actually just the RHF ONV, since we're using the canonical RHF orbitals) and the corresponding orbital space.
+    //const auto reference_onv = GQCP::SpinUnresolvedONV::GHF(2 * K, N, rhf_parameters.spinOrbitalEnergiesBlocked());
+    //const auto orbital_space = reference_onv.orbitalSpace();
+    // Use a manually-made orbital space (#FIXME in develop).
+    const auto orbital_space = GQCP::OrbitalSpace({0, 1, 2, 3, 4, 7, 8, 9, 10, 11}, {5, 6, 12, 13});  // occupied and virtual indices
+
+
+    BOOST_REQUIRE(orbital_space.numberOfOrbitals() == M);
+
+
+    // Initialize an environment suitable for CCD.
+    auto environment_ccd = GQCP::CCSDEnvironment<double>::PerturbativeCCD(g_sq_hamiltonian, orbital_space);
+    auto environment_ccsd_ref = GQCP::CCSDEnvironment<double>::PerturbativeCCSD(g_sq_hamiltonian, orbital_space);
+
+    // Functional step that sets the T1-amplitudes to zero, needed for our reference CCD solver.
+    //GQCP::FunctionalStep<GQCP::CCSDEnvironment<double>> set_T1_zero([GQCP::OrbitalSpace orbital_space](GQCP::CCSDEnvironment<double> environment_ccsd_ref){
+    //   environment_ccsd_ref.t1_amplitudes.back() = GQCP::T1Amplitudes<double>(orbital_space.initializeRepresentableObjectFor<double>(GQCP::OccupationType::k_occupied, GQCP::OccupationType::k_virtual), orbital_space);
+    //    }, "Set the T1-amplitudes to zero.");
+    GQCP::FunctionalStep<GQCP::CCSDEnvironment<double>> set_T1_zero([](GQCP::CCSDEnvironment<double>& environment_ccsd_ref){
+        const auto& orbital_space = environment_ccsd_ref.t1_amplitudes.back().orbitalSpace();
+        environment_ccsd_ref.t1_amplitudes.back() = GQCP::T1Amplitudes<double>(environment_ccsd_ref.t1_amplitudes.back().orbitalSpace().initializeRepresentableObjectFor<double>(GQCP::OccupationType::k_occupied, GQCP::OccupationType::k_virtual), orbital_space);
+    }, "Set the T1-amplitudes to zero.");
+
+    // Prepare the CCD solver and optimize the CCD model parameters.
+    auto solver_ccd = GQCP::CCDSolver<double>::Plain();
+    const auto ccd_qc_structure = GQCP::QCMethod::CCD<double>().optimize(solver_ccd, environment_ccd);
+
+    // Prepare the CCSD reference solver and optimize the CCSD model parameters but with T1-amplitudes all 0.
+    auto solver_ref = GQCP::CCSDSolver<double>::Plain();
+    solver_ref.insert(set_T1_zero, 2);
+    const auto ref_qc_structure = GQCP::QCMethod::CCSD<double>().optimize(solver_ref, environment_ccsd_ref);
+
+    const auto ccd_correlation_energy = ccd_qc_structure.groundStateEnergy();
+    const double ref_ccd_correlation_energy = ref_qc_structure.groundStateEnergy();
+
+    BOOST_CHECK(std::abs(ccd_correlation_energy - ref_ccd_correlation_energy) < 1.0e-08);
+}
```

Comment:
Could you also add a test that checks if the T2-amplitudes found by both solvers are equal?